src/app/task-list/task-list.component.ts

```diff
@@ -1,26 +1,55 @@
 import { Component, OnInit } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { TaskService } from '../service/task.service';
-import { Task } from '../task';
+import { Task } from '../models/task.model';
+import { Subject, takeUntil } from 'rxjs';
+import { Router, RouterLink } from '@angular/router';
+import { HttpClientModule } from '@angular/common/http';
 
 @Component({
   selector: 'tasko-task-list',
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, RouterLink, HttpClientModule],
   templateUrl: './task-list.component.html',
   styleUrl: './task-list.component.scss'
 })
 export class TaskListComponent implements OnInit {
   tasks: Task[] = [];
+  private destroy$: Subject<void>= new Subject<void>;
 
-  constructor(private taskService: TaskService) {}
+  constructor(private taskService: TaskService, private router :Router) {
+    // console.log(data);
+  }
 
   ngOnInit() {
-    
-    this.tasks = this.taskService.getTasks();
+    this.getTasks();
+    // this.tasks = this.taskService.getTasks();
+  }
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
   }
 
-  onDeleteTask(index: number) {
-    this.taskService.deleteTask(index);
+  getTasks(){
+    this.taskService.getAllTasks().pipe(takeUntil(this.destroy$)).subscribe({
+      next: (response:Task[])=>{
+        this.tasks=response;
+      },
+      error: (error:Error) =>{
+        console.error(error);
+      }
+    })
+  }
+  deleteATask(taskId: string):void{
+    this.taskService.deleteATask(taskId).pipe(takeUntil(this.destroy$)).subscribe(()=>{
+      this.getTasks();
```

Comment:
don't use a subscription inside of a subscription, if you delete a task (i assume this is in the task list page) you can return a list of the new tasks (without the deleted task)